src/middlewares.ts

```diff
@@ -16,13 +16,22 @@ const hashPattern = /(?:^|.*?\.)hash-([a-f0-9]+)\./;
 function assembleSubdomainUrlForHash(req: any, commitHash: CommitHash) {
   const protocol = req.secure || req.headers.host.indexOf( 'calypso.live' ) > -1 ? "https" : "http";
 
+  const query =
+    Object.keys( req.query )
+    .reduce(
```

Comment:
i think the `querystring` module will do this for you

Reply:
https://nodejs.org/api/querystring.html#querystring_querystring_stringify_obj_sep_eq_options

i think you're removing the `branch` key and then serializing the remaining query?

Reply:
> i think you're removing the `branch` key and then serializing the remaining query?

That was the intention, but it didn't work as expected. See the resulting redirections from #84:

- `/?branch=master` -> `/?`
- `/?branch=master&flags=jetpack/checklist` -> `/?flags=undefined`